src/Kiota.Builder/Extensions/OpenApiUrlTreeNodeExtensions.cs

```diff
@@ -38,7 +40,9 @@ public static class OpenApiUrlTreeNodeExtensions {
         private static string CleanupParametersFromPath(string pathSegment) {
             if((pathSegment?.Contains(requestParametersChar) ?? false) ||
                 (pathSegment?.Contains(requestParametersSectionChar) ?? false))
-                return PathParametersRegex.Replace(pathSegment, requestParametersMatchEvaluator)
+                return PathParametersRegex.Replace(
+                                            AtSignPathParameterRegex.Replace(pathSegment, "={$1}"),
```

Comment:
This will work for scenarios with a single parameter. What about for 2 or more parameters?

Reply:
It's a regex replacement with the global flag (regex are global by default in dotnet), so it'll replace every instance. [See this fiddle](https://dotnetfiddle.net/QACJXK]